File: python/y2019/day02.py
from aoc.core import solve, AocInput
from aoc.intcode import IntCodeMachine


def part1(input: AocInput) -> int:
    icm = IntCodeMachine.init(input.text)
    return IntCodeMachine.run(icm, noun=12, verb=2)


def part2(input: AocInput) -> int:
    icm = IntCodeMachine.init(input.text)
    return IntCodeMachine.gravityAssist(icm)


def main() -> None:
    YEAR = "y2019"
    DAY = "Day02"
    print(f"Running year {YEAR.replace('y', '')} day {DAY.replace('Day', '')}")
    solve(YEAR, DAY, part1, expected=3790645, skipTest=True)
    solve(YEAR, DAY, part2, expected=6577, skipTest=True)

File: dotnet/2019/AdventOfCode.Y2019/Day02.cs
using AdventOfCode.Lib;

namespace AdventOfCode.Y2019;

[AocPuzzle(2019, 2, "1202 Program Alarm")]
public sealed class Day02 : IAocDay<int>
{
    public static int Part1(AocInput input) =>
        input.Text
        | IntCodeMachine.Init
        | (icm => IntCodeMachine.Run(icm, 12, 2));

    public static int Part2(AocInput input) =>
        input.Text
        | IntCodeMachine.Init
        | (icm => IntCodeMachine.GravityAssist(icm, 100, 100));
}